src/middleware/validate/validateBooleanMiddleware.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { RequestHandler } from 'express';
+import { BadRequest } from 'http-errors';
+
+// Query Params used in controllers
+const queryParams = [
+	'eventDocs', // BlocksController
+	'extrinsicDocs', // BlocksController
+	'finalized', // BlocksController
+	'denominated', // AccountsBalanceController
+	'unclaimedOnly', // AccountsStakingPayoutsController
+	'actions', // BlocksTraceController
+	'includeFee', // NodeTransactionPoolController
+	'adjustMetadataV13', // PalletsStorageController
+	'currentLeaseHolders', // ParasController
+	'noMeta', // TransactionMaterialController
+];
```

Comment:
I'm not a huge fan of this sort of "global list". What if "actions" appears in some other thing in the future and is not a boolean, there?

It makes me wonder more about the value of such a middleware?

It also makes me think of npm packages like Runtypes, which allow you to validate all of the type info out of some JS object.

I geuss I'm wondering; what is actually the benefit of such a middleware?

Reply:
> I'm not a huge fan of this sort of "global list". What if "actions" appears in some other thing in the future and is not a boolean, there?

Im not the biggest fan either but its necessary given the contraints.
So the middleware is only ran if you add it to the endpoint's route itself, so it would have to be explicitly added and would error. It would be very hard to introduce a crossover. 

> It makes me wonder more about the value of such a middleware?

The value of this is centralizing an error for invalid boolean query params. It doesn't do any sanitizing or mutating. Just error handling.

Reply:
Also this helps with cleaning up a lot of the controllers for error handling.

Reply:
Removed the global list.